module-db/Database/sqlite3vfs.cpp

```diff
@@ -158,19 +158,16 @@ struct EcophoneFile
     sqlite3_int64 iBufferOfst; /* Offset in file of zBuffer[0] */
 
     /* Current state */
-    long _size = -1;
     long _pos  = -1;
 
     /* Basic operations */
     auto size()
     {
-        if (_size < 0) {
-            _pos = std::ftell(fd);
-            std::fseek(fd, 0, SEEK_END);
-            _size = std::ftell(fd);
-            std::fseek(fd, _pos, SEEK_SET);
-        }
-        return _size;
+        _pos = std::ftell(fd);
```

Comment:
I wonder if we need `_pos` at all? In that case invoking `size()` introduces some side effects :(

Reply:
These variables (`_pos_` & `_size_`) were probably created to speed up the algorithm. With large databases, it actually worked faster, but the `_size` variable was definitely introducing errors. What side effects are you talking about?

Reply:
Invoking `size()` internally updates local variable `_pos` which is available throughout the module which, for me is a side effect. Function does something more than specified in declaration. 

Anyway, let's leave it like that. Removing this would probably introduce some further changes which we currently do now want.

Please, feel free to resolve this conversation :)